src/confdb/confdb.c

```diff
@@ -2423,3 +2443,198 @@ int confdb_certmap_to_sysdb(struct confdb_ctx *cdb,
+    char** auxi_enabled_list = NULL;
+    char** auxi_disabled_list = NULL;
+    TALLOC_CTX* tmp_ctx = NULL;
+    int enabled = -1;
+
+    if (! explicit_enabled && ! explicit_disabled) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    tmp_ctx = talloc_new(NULL);
+    if (! tmp_ctx) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = confdb_list_all_domain_names(tmp_ctx, cdb, &all_domains);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Failed retrieving all domain name "
+              "list, error [%d], description '%s'\n",
+              ret, strerror(ret));
+        goto done;
+    }
+
+    for (int idx = 0; all_domains[idx]; idx++) {
+        char* section = talloc_asprintf(tmp_ctx, CONFDB_DOMAIN_PATH_TMPL, all_domains[idx]);
+        values = NULL;
+        ret = confdb_get_param(cdb, tmp_ctx, section, CONFDB_DOMAIN_ENABLED, &values);
+        if (ret != EOK && ret != ENOENT) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Failed retrieving '%s' attribute in '%s' section; "
+                  "error [%d], description '%s'\n",
+                  CONFDB_DOMAIN_ENABLED, section,
+                  ret, strerror(ret));
+            goto done;
+        }
+
+        if (ret == ENOENT || ! values || ! values[0]) {
+            enabled = -1;
+        } else if (0 == strcasecmp(values[0], "true")) {
+            enabled = 1;
+        } else if (0 == strcasecmp(values[0], "false")) {
+            enabled = 0;
+        } else {
+            enabled = -1;
+        }
+
+        if (enabled == -1) {
+            continue;
+        } else if (enabled == 1) {
+            if (explicit_enabled) {
+                ret = add_string_to_list(tmp_ctx, all_domains[idx], &auxi_enabled_list);
+                if (ret != EOK) {
+                    DEBUG(SSSDBG_OP_FAILURE,
+                          "Failed retrieving '%s' attribute in '%s' section; "
+                          "error [%d], description '%s'\n",
+                          CONFDB_DOMAIN_ENABLED, section,
+                          ret, strerror(ret));
+                    goto done;
+                }
+            }
+        } else if (enabled == 0) {
+            if (explicit_disabled) {
+                ret = add_string_to_list(tmp_ctx, all_domains[idx], &auxi_disabled_list);
+                if (ret != EOK) {
+                    DEBUG(SSSDBG_OP_FAILURE,
+                          "Failed retrieving '%s' attribute in '%s' section; "
+                          "error [%d], description '%s'\n",
+                          CONFDB_DOMAIN_ENABLED, section,
+                          ret, strerror(ret));
+                    goto done;
+                }
+            }
+        }
+    }
+
+    /* Retain the lists */
+    if (explicit_enabled) {
+        talloc_steal(ctx, auxi_enabled_list);
+        *explicit_enabled = auxi_enabled_list;
+    }
+    if (explicit_disabled) {
+        talloc_steal(ctx, auxi_disabled_list);
```

Comment:
If no enabled attribute is found equals to "false", auxi_disabled_list is NULL, so nothing to be retained; Check condition before retain.